### PR TITLE
feat: add support for danish language

### PR DESCRIPTION
**Pull request by Franta1205**
I added Danish support, tested and is currently actively used by Danish-speaking people.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,11 @@
 ### Features
   * Your contribution here.
 
+## 0.11.13 (May 2, 2024)
```
Comment (dblock) [inline CHANGELOG.md:6]: Remove this, we bump during release.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,11 @@
...
+
+### Features
```
Comment (dblock) [inline CHANGELOG.md:8]: Put this above "Your contribution here" above.

File: numbers_and_words.gemspec [hunk 3]
```diff
@@ -2,16 +2,16 @@
 # DO NOT EDIT THIS FILE DIRECTLY
 # Instead, edit Jeweler::Tasks in Rakefile, and run 'rake gemspec'
 # -*- encoding: utf-8 -*-
-# stub: numbers_and_words 0.11.12 ruby lib
+# stub: numbers_and_words 0.11.13 ruby lib
 
 Gem::Specification.new do |s|
   s.name = "numbers_and_words".freeze
-  s.version = "0.11.12"
+  s.version = "0.11.13"
```
Comment (dblock) [inline numbers_and_words.gemspec:9]: Revert any version changes, please.

File: numbers_and_words.gemspec [hunk 4]
```diff
@@ -218,13 +223,21 @@ Gem::Specification.new do |s|
...
-  s.add_development_dependency(%q<rake>.freeze, ["~> 12"])
+  if s.respond_to? :add_runtime_dependency then
+    s.add_runtime_dependency(%q<i18n>.freeze, ["<= 2"])
+    s.add_development_dependency(%q<jeweler>.freeze, ["~> 2"])
+    s.add_development_dependency(%q<rake>.freeze, ["~> 12"])
+  else
+    s.add_dependency(%q<i18n>.freeze, ["<= 2"])
+    s.add_dependency(%q<jeweler>.freeze, ["~> 2"])
+    s.add_dependency(%q<rake>.freeze, ["~> 12"])
+  end
```
Comment (dblock) [inline numbers_and_words.gemspec:241]: This seems unrelated, you should revert these parts. If there's a reason to make this dynamic, let's put this in another PR.

File: spec/numbers_and_words/i18n_spec.rb [hunk 5]
```diff
@@ -5,7 +5,7 @@
 describe NumbersAndWords::I18n do
   describe '.local_language' do
     it 'checks validity of given language and return it as symbol' do
-      expect(described_class.local_language(:de)).to eq(:de)
+      expect(described_class.local_language(:en)).to eq(:en)
```
Comment (dblock) [inline spec/numbers_and_words/i18n_spec.rb:8]: Why this change?

**Review by dblock — CHANGES_REQUESTED** [review]
It looks good! Undo the version changes please, and add `.idea` to `.gitignore` and remove the folder from the PR, it came along for the ride.

File: CHANGELOG.md (hunk 1 above)
Comment (Franta1205) [reply CHANGELOG.md]: done

File: numbers_and_words.gemspec (hunk 4 above)
Comment (Franta1205) [reply numbers_and_words.gemspec]: done

File: numbers_and_words.gemspec (hunk 3 above)
Comment (Franta1205) [reply numbers_and_words.gemspec]: done

**Comment by dblock** [comment]
Looks like CI is failing, take a look?

Since this _is_ a new feature we should increment the non-patch version from 0.11.13 to 0.12.0. Change the number wherever you find it?

**Comment by Franta1205** [comment]
> Looks like CI is failing, take a look?
> 
> Since this _is_ a new feature we should increment the non-patch version from 0.11.13 to 0.12.0. Change the number wherever you find it?

Yes still failing will take look.

**Comment by Franta1205** [comment]
> I think this is as simple as `bundle exec rake` instead of `rake` in CI. I upgraded rubocop in #198, will merge that, rebase on top of it?

should I rebase then?

**Comment by Franta1205** [comment]
> I think this is as simple as `bundle exec rake` instead of `rake` in CI. I upgraded rubocop in #198, will merge that, rebase on top of it?

done

**Comment by dblock** [comment]
Rubocop is complaining, run `rubocop -a`. Check that `rubocop` passes.

Double check that the specs include Danish. There are some things like `FLOAT_CAPABLE_LANGUAGES` that you may need to add it to as well.

**Comment by Franta1205** [comment]
> rubocop -a

I forgot to update the specs after solving some issues. Now all the tests pass correctly locally. Should be fine now. Sorry, I missed that.

**Comment by dblock** [comment]
> > rubocop -a
> 
> I forgot to update the specs after solving some issues. Now all the tests pass correctly locally. Should be fine now. Sorry, I missed that.

No stress, but RuboCop still unhappy (https://github.com/kslazarev/numbers_and_words/actions/runs/9098488656/job/25014903295?pr=197). Try `rubocop -a`, really :)

**Comment by Franta1205** [comment]
> > > rubocop -a
> > 
> > 
> > I forgot to update the specs after solving some issues. Now all the tests pass correctly locally. Should be fine now. Sorry, I missed that.
> 
> No stress, but RuboCop still unhappy (https://github.com/kslazarev/numbers_and_words/actions/runs/9098488656/job/25014903295?pr=197). Try `rubocop -a`, really :)

yep, let's see if it's fine now

**Comment by dblock** [comment]
Merged, thank you!

I'd like to get #108 in and make a release, maybe you can help with that one?

**Comment by Franta1205** [comment]
> Merged, thank you!
> 
> I'd like to get #189 in and maybe #150 and make a release, maybe you can help with that one?

Yes no prob will have look in to it

**Comment by jlduran** [comment]
Hello @Franta1205, sorry to bother almost a year after this pull request.
This gem attempts to achieve 100% code coverage across all languages, as it helps us maintain it better.
Coveralls says it is currently missing the following line of code:
https://github.com/kslazarev/numbers_and_words/blob/a5c28faabbe4d72eb578b9f65f41db5e8af05d13/lib/numbers_and_words/strategies/figures_converter/languages/da.rb#L49
Would you mind adding an example or replying if the following patch is grammatically correct:

```diff
--- a/spec/numbers_and_words/integer/fixture_examples/da.yml
+++ b/spec/numbers_and_words/integer/fixture_examples/da.yml
@@ -3,3 +3,11 @@ to_words:
     0: nul
     1: et
     9: ni
+  common gender:
+    options:
+      :gender: common
+    1000000: en million
+  neuter gender:
+    options:
+      :gender: neuter
+    1000000: en million
```

Thank you!

**Comment by Franta1205** [comment]
> Hello @Franta1205, sorry to bother almost a year after this pull request. This gem attempts to achieve 100% code coverage across all languages, as it helps us maintain it better. Coveralls says it is currently missing the following line of code:
> 
> https://github.com/kslazarev/numbers_and_words/blob/a5c28faabbe4d72eb578b9f65f41db5e8af05d13/lib/numbers_and_words/strategies/figures_converter/languages/da.rb#L49
> 
> 
> Would you mind adding an example or replying if the following patch is grammatically correct:
> ```diff
> --- a/spec/numbers_and_words/integer/fixture_examples/da.yml
> +++ b/spec/numbers_and_words/integer/fixture_examples/da.yml
> @@ -3,3 +3,11 @@ to_words:
>      0: nul
>      1: et
>      9: ni
> +  common gender:
> +    options:
> +      :gender: common
> +    1000000: en million
> +  neuter gender:
> +    options:
> +      :gender: neuter
> +    1000000: en million
> ```
> 
> Thank you!

Hey,

i will add examples. Some time next week.
